Simple frontend

kabachook opened the pull request:
Self-explanatory

kabachook committed:
Simple frontend

gohabereg commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,72 @@
+let shouldSearch = false;
+
+const searchDomain = async (domain) => {
+  const response = await fetch('http://localhost:3000' + DOMAIN_ENDPOINT + domain);
```
Надо разхардкодить урл

gohabereg commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,72 @@
+const processQuery = async (domain) => {
+  const resp = await searchDomain(domain);
+
+  console.log(resp);
```
Лог убрать

gohabereg commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,72 @@
+  const resp = await searchDomain(domain);
+
+  console.log(resp);
+  const resultElement = document.querySelector(RESULT_DIV);
```
Если у тебя у этих элементов есть id, то лучше искать по id

gohabereg commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,72 @@
+  const searchElement = document.querySelector(SEARCH_INPUT);
+
+  for (let [tld, data] of Object.entries(resp)) {
+    if (Object.keys(data).indexOf('available') !== -1 && data['available'] === true) {
```
Можно просто вот так. Если такого ключа нет, то `data.available === undefined`
```suggestion
    if (data.available === true) {
```

gohabereg commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,72 @@
+  return true;
+};
+
+onload = _ => {
```
```suggestion
onload = () => {
```

gohabereg commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,72 @@
+      child.innerHTML = `${searchElement.value}.${tld}`;
+
+      resultElement.appendChild(child);
+      // resultElement.appendChild(document.createElement('break'));
```
Лишние строки убирай

gohabereg commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,72 @@
+
+  document.querySelector('input#search-input').addEventListener('keydown', (event) => {
+    console.log('keydown called');
+    if (event.keyCode === 13 || event.keyCode === 10) {
```
Коды клавиш надо расхардкодить

gohabereg commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,72 @@
+const SEARCH_THRESHHOLD = 1000;
```
Все методы в этом файле надо или оформить в класс, или обернуть в замыкание

gohabereg committed:
Fix _ to () in array function

Co-Authored-By: kabachook <beltyukovdanil@gmail.com>

gohabereg committed:
Remove stupid check

Co-Authored-By: kabachook <beltyukovdanil@gmail.com>

kabachook committed:
Code refactoring

- Removed form element. so no need to preventDefault in js
- Now searching elements by id
- Some small fixes

kabachook committed:
Serve frontend by express for dev

nikmel2803 committed:
add closure

nikmel2803 committed:
frontend flexbox fix

nikmel2803 committed:
frontend flexbox refactor

nikmel2803 committed:
add eslint config in frontend

nikmel2803 committed:
delete onload handler

nikmel2803 committed:
refactor

neSpecc commented on frontend/src/css/main.css:
```diff
@@ -0,0 +1,38 @@
+html,
+body {
+    font-family: 'Inconsolata', monospace;
```
надо использовать системные шрифты

neSpecc commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,62 @@
+(function () {
+  const API_URI = 'http://localhost:3000';
```
надо задокументировать код, описать все методы, константы, параметры и тд

neSpecc commented on frontend/src/js/main.js:
```diff
@@ -0,0 +1,62 @@
+
+  let shouldSearch = false;
+
+  const searchDomain = async (domain) => {
```
async нельзя использовать без специальных полифиллов

neSpecc commented on frontend/src/index.html:
```diff
@@ -0,0 +1,21 @@
+
+<body>
+
+<h1>Hello, World!</h1>
```
нет табуляции

neSpecc commented on frontend/package.json:
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "frontend",
```
тут указывается название проекта

kabachook commented:
Moved to github.com/codex-team/domain-checker
